src/ordinal.rs

```diff
@@ -107,9 +107,17 @@ impl Ordinal {
 
     let cycle_start_epoch = cycle_number * CYCLE_EPOCHS;
 
-    let cycle_progression = period_offset
-      .checked_sub(epoch_offset % PERIOD_BLOCKS)
```

Comment:
`epoch_offset` is a number betwenn 0 and 210_000 that represents number of blocks mined since last Halving.

`epoch_offset % PERIOD_BLOCKS` gives us the `period_offset`, but only if we are at the beginning of a cycle (in the first of six halving epochs).

The correct period_offset:
`period_offset = cycle_progression % PERIOD_BLOCKS`

I'm a little bit confused now and will look at this tomorrow.